Source/WTF/wtf/text/Base64.cpp

```diff
@@ -193,9 +194,9 @@ template<typename T> static std::optional<Vector<uint8_t>> base64DecodeInternal(
                 if (equalsSignCount)
                     return std::nullopt;
                 destination[destinationLength++] = decodedCharacter;
-            } else if (!options.contains(Base64DecodeOptions::IgnoreSpacesAndNewLines) || (!isLatin1(ch) || !isASCIISpace(ch) || (options.contains(Base64DecodeOptions::DiscardVerticalTab) && ch == '\v'))) {
+            // isASCIISpace includes U+000B which https://infra.spec.whatwg.org/#forgiving-base64 excludes
+            } else if (mode != Base64DecodeMode::DefaultValidatePaddingAndIgnoreWhitespace || !isASCIISpace(ch) || ch == '\v')
```

Comment:
Seems a tiny bit sad to have to write this out like this. I wish we could take the infrastructure specification’s version of whitespace and put a sensibly named function somewhere in `WTF` for it. Then `isHTMLSpace` could just call it, or even eventually be removed.

Reply:
I’ll go a step further and say that we should factor this expression into an inline function for this definition of whitespace, even if the function is local to this file. Call it `isForgivingBase64Whitespace` and then the comment makes even more sense. Follows my principle of “use the fact that you had to write a comment to inform how you write the code” and in some cases one can even remove a comment entirely if you name carefully.

Reply:
One last thought. I find that making the correct local function makes it easier to refactor and share later.

Reply:
Thanks @darinadler! If #13077 looks good we can land that first and then I will rebase this.